api/src/database/database.py

```diff
@@ -151,6 +151,8 @@ def select(
             return results
         except Exception as e:
             logging.error(f"SELECT query failed with exception: \n{e}")
+            if global_session is not None:
+                global_session.rollback()
```

Comment:
If the global session is shared, what is the behavior when multiple requests are processed at the same time?

Reply:
Opened #293 to address this problem